### ISSUE TITLE
chained-comparison warning with indexes (not chainable)

### ISSUE THREAD

**Issue by SubaruArai** [PyCQA/pylint#6599]
### Bug description

```python3
#!/usr/bin/env python3
# dummy.py


dummy_list = [1, 2]
if dummy_list[0] < 0 and dummy_list[1] >= 0:  # chain warning here
    print("dummy")
```

This cannot be chained, but pylint will warn to chain.

### Command used

```shell
`pylint dummy.py`
```

### Pylint output

```shell
************* Module dum
dum.py:1:0: C0114: Missing module docstring (missing-module-docstring)
dum.py:6:3: R1716: Simplify chained comparison between the operands (chained-comparison)

------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 3.33/10, +0.00)
```

### Expected behavior

pylint doesn't warn about it.

### Pylint version

```shell
pylint 2.13.8
astroid 2.11.5
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04 LTS

**Comment by DanielNoord** [comment]
Thanks for the report! I do wonder though, wouldn't this work as well:
```python
#!/usr/bin/env python3
# dummy.py


dummy_list = [1, 2]
if dummy_list[0] < 0 <= dummy_list[1]:
    print("dummy")
```

**Comment by SubaruArai** [comment]
@DanielNoord  Damn silly me! I didn't notice, thanks for pointing out!
